flink-python/pyflink/fn_execution/coder_impl.py

```diff
@@ -24,57 +24,71 @@
 from pyflink.table import Row
 
 
+def generate_null_mask_search_table():
+    """
+    Each bit of one byte represents if the column at the specified position is None or not, e.g.
+    0x84 represents the first column and the sixth column are None.
+    """
+    num = 256
+    null_mask = []
+    for b in range(num):
+        every_num_null_mask = [(b & 0x80) > 0, (b & 0x40) > 0, (b & 0x20) > 0, (b & 0x10) > 0,
+                               (b & 0x08) > 0, (b & 0x04) > 0, (b & 0x02) > 0, (b & 0x01) > 0]
+        null_mask.append(tuple(every_num_null_mask))
+
+    return tuple(null_mask)
+
+
 class FlattenRowCoderImpl(StreamCoderImpl):
+    null_mask_search_table = generate_null_mask_search_table()
```

Comment:
What about make it an instance variable as it's faster than static variable: https://stackoverflow.com/questions/2714573/instance-variables-vs-class-variables-in-python